extension-manifest-v2/app/content-scripts/youtube.js

```diff
@@ -0,0 +1,26 @@
+import { showIframe, closeIframe } from '@ghostery/ui/iframe';
+
+// Based on https://github.com/AdguardTeam/AdguardFilters/blob/e5ae8e3194f8d18bdcc660d4c42282e4a96ca5b9/AnnoyancesFilter/Popups/sections/antiadblock.txt#L2044
+const ADBLOCKER_WALL_SELECTORS = [
+	'ytd-enforcement-message-view-model > div.ytd-enforcement-message-view-model',
+];
+
+let isShown = false;
+
+chrome.storage.local.get(['youtube_dont_show_again'], (storage) => {
+	if (storage.youtube_dont_show_again || chrome.extension.inIncognitoContext) {
+		return;
+	}
+
+	window.addEventListener('yt-navigate-start', () => {
+		isShown = false;
+		closeIframe();
+	}, true);
+
+	setInterval(() => {
+		if (!isShown && document.querySelectorAll(ADBLOCKER_WALL_SELECTORS).length > 0) {
```

Comment:
Detail, but I think that should work as well:

 document.querySelector(ADBLOCKER_WALL_SELECTORS)
 instead of
 document.querySelectorAll(ADBLOCKER_WALL_SELECTORS).length > 0